referenceframe/worldstate.go

```diff
@@ -14,27 +14,27 @@ const unnamedWorldStateGeometryPrefix = "unnamedWorldStateGeometry_"
 
 // WorldState is a struct to store the data representation of the robot's environment.
 type WorldState struct {
-	obstacleNames map[string]bool
-	obstacles     []*GeometriesInFrame
-	transforms    []*LinkInFrame
+	ObstacleNamesInner map[string]bool      `json:"obstacle_names"`
```

Comment:
Making these public makes a previously-immutable data structure now extremely editable. I don't look forwards to the years and years of bugs that will be introduced by people editing things that get passed around by pointer and causing myriad issues.

I know that struct members need to be public to unmarshal from JSON; if that's the driving force here, we should have an intermediate data structure specifically for this purpose, like GeometryConfig in `spatialmath`.